include/zephyr/sensing/sensing_sensor.h

```diff
@@ -0,0 +1,594 @@
+ *
+ * @param data_ready Enable/disable the data ready mode. Default:disabled
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int sensing_sensor_set_data_ready(
+		const struct device *dev, bool data_ready);
+
+/**
+ * @}
+ */
+
+/**
+ * @brief Sensor Callbacks
+ * @addtogroup sensing_sensor_callbacks
+ * \{
+ */
+
+/**
+ * @brief Sensor initialize callback.
+ *
+ * Sensor can initialize it's runtime context in this callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param info The sensor instance's constant information.
+ *
+ * @param reporter_handles The reporters handles for this sensor, NULL for physical sensors.
+ *
+ * @param reporters_count The number of reporters, zero for physical sensors.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*sensing_sensor_init_t)(
+		const struct device *dev, const struct sensing_sensor_info *info,
+		const int *reporter_handles, int reporters_count);
+
+/**
+ * @brief Sensor's de-initialize callback.
+ *
+ * Sensor can release it's runtime context in this callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*sensing_sensor_deinit_t)(
+		const struct device *dev);
+
+/**
+ * @brief Sensor reset callback.
+ *
+ * Sensor can reset its runtime context in this callback to default values without resources
+ * release and re-allocation.
+ *
+ * Its very useful for a virtual sensor to quickly reset its runtime context to a default state.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*sensing_sensor_reset_t)(
+		const struct device *dev);
+
+/**
+ * @brief Sensor read sample callback.
+ *
+ * Only physical sensor need implement this callback.
+ * Physical sensor can fetch sample data from sensor device in this callback
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param buf Sensor subsystem runtime allocated buffer, and passed its pointer
+ * to this sensor for store fetched sample.
+ *
+ * @param size The size of the buffer in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*sensing_sensor_read_sample_t)(
+		const struct device *dev,
+		void *buf, int size);
+
+/**
+ * @brief Sensor process data callback.
+ *
+ * Only virtual sensor need implement this callback.
+ * Virtual sensor can receive reporter's data and do fusion computing
+ * in this callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param reporter The reporter handle who delivered this sensor data
+ *
+ * @param buf The buffer stored the reporter's sensor data.
+ *
+ * @param size The size of the buffer in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*sensing_sensor_process_t)(
+		const struct device *dev,
+		int reporter,
```

Comment:
Can the `reporter` parameter be managed as a `struct device *` instead?  I assume this points to one of the entries from the `reporters` property, but the implementation is missing.

Reply:
We can't use `struct device *` for `reporter` here. 

Since here we need the open `handle` with the reporter sensor, not the reporter sensor device itself. 

Since same report sensor can report to different client sensors at same time, we use the `handle` for the report relationship between a client sensor and a report sensor, each `handle` will have different configurations to report sensor by this client sensor. 

And will use typedef of `handle`, then will be more clear, see https://github.com/zephyrproject-rtos/zephyr/pull/55389/files#r1197561702